PSReadLine/ReadLine.cs

```diff
@@ -201,6 +203,12 @@ internal static PSKeyInfo ReadKey()
                     if (handleId != WaitHandle.WaitTimeout && handleId != EventProcessingRequested)
                         break;
 
+                    if (_handleIdleOverride != null)
+                    {
+                        _handleIdleOverride();
+                        continue;
```

Comment:
Won't this continue hard stop event processing/cancellation handling?

Reply:
So the current wait handle will handle cancellation -- that happens above the delegate call. We also own the cancellation token, so we have a lot of control there.

Event processing is what we need to prevent, because if we invoke PSRL directly as a delegate with no pipeline running above it, event handling will fail. PSES owns the pipeline and pretty much everything else at this point, so I think it's up to us to handle events in our own way.

Reply:
> we invoke PSRL directly as a delegate with no pipeline running above it, event handling will fail

Even if you set `Runspace.DefaultRunspace` before calling PSRL? I was pretty certain `Create(RunspaceMode.CurrentRunspace)` worked as long as that's set.  It's definitely supposed to at least, there's code for it.

Or is it failing some other way?

> so I think it's up to us to handle events in our own way.

Yeah that's fine if it comes to that, just a shame to implement the same logic in two places I suppose.

Reply:
Hmmm, well in fact looking into this, it may be that I'm not quite invoking the delegate the right way (i.e. not on the pipeline thread). I'll look into it and get back to you, because you may well be right! (Will keep this in draft until kinks like this are resolved)

Reply:
Just looked into this, and yeah the delegate is marshalled back to the pipeline thread and called there (and we have `Runspace.DefaultRunspace` set). I'll see if I can try again, but that's the whole reason for the `continue`, since PSRL's default event handling didn't work for us in the top-level pipeline (I spent ages debugging it...)

Reply:
If the event handling doesn't work, make sure `TabExpansion2` and script based custom key handlers still work.

I worry that if PSRL's event handling doesn't work, I'm not 100% sure how we're going to handle that better in PSES you know?  If we lose the ability to process events it might be better to ditch the delegate.